src/components/RadioForm.astro

```diff
@@ -0,0 +1,53 @@
+---
+import { t } from '../i18n/server';
+
+interface Props {
+    actionUrl: string;
+    idPrefix: string;
+    options: Record<string, string>;
+    value: string | null | undefined;
+}
+
+const { idPrefix, actionUrl, options, value } = Astro.props;
+---
+
+<div class="col66 col100-mobile">
+    <div class="radio-group radio-group-vertical radio-group-padded">
+        <form method="POST" id={`${idPrefix}-form`} action={actionUrl} class="astro-radio-form">
+            {
+                Object.keys(options).map((option) => (
+                    <div class="radio-wrapper">
+                        <input
+                            id={`${idPrefix}-radio-${option}`}
+                            type="radio"
+                            name="answer"
+                            class="form-element"
+                            value={option}
+                            checked={option === value}
+                            required="true"
+                        />
+                        <label class={`radio-label `} for={`${idPrefix}-radio-${option}`}>
+                            {options[option]}
+                        </label>
+                    </div>
+                ))
+            }
+            <button type="submit" class="button sr-only">{t('common', 'continue')}</button>
+        </form>
+    </div>
+</div>
+
+<script>
+    const forms = document.getElementsByClassName('astro-radio-form') as HTMLCollectionOf<HTMLFormElement>;
```

Comment:
Isn't this run for every form on the page, causing multiple event handlers to be registered if there are multiple forms? You know the ID of 'your' form, you only need to add it to that one.